### PR TITLE
Rename and re-organize reviewer groups

### LINKED ISSUE #6611: Reorganize reviewer groups
We need to make some changes to our permission groups for reviewers to keep up with recent updates:

* [ ] Rename `Add-on Reviewers` as `Reviewers: Legacy`, and only give them `Addons:Review` permission.
* [ ] Rename `Unlisted Add-on Reviewers` as `Reviewers: Unlisted`.
* [ ] Rename `Persona Reviewers` as `Reviewers: Themes`. 
* [ ] Figure out if the `SeniorPersonasTools:View` permission is used for anything. If so, add that permission to the `Reviewers: Themes` group.
* [ ] Remove the `Senior Personas Reviewers` group.
* [ ] Remove the `Senior Add-on Reviewers` group.
* [ ] Create a `Reviewers: Content` group with the permission `Addons:ContentReview`.
* [ ] Create a `Reviewers: Add-ons` group with the permission `Addons:PostReview`.
* [ ] Create a `Reviewers: Moderators` group with the permission `Reviews:Edit`. If possible, copy over all users in the `Reviewers: Legacy` group. Otherwise we can do this manually.

### PR DESCRIPTION
**Pull request by diox**
- Rename `Add-on Reviewers` as `Reviewers: Legacy`, and only give them `Addons:Review` permission.
- Create Unlisted Add-on Reviewers group named `Reviewers: Unlisted`.
- Rename `Persona Reviewers` as `Reviewers: Themes`.
- Remove the `Senior Personas Reviewers` group and `SeniorPersonasTools:View` permission (all Personas reviewer tools are available to all personas reviewers).
- Remove the `Senior Add-on Reviewers` group.
- Create a `Reviewers: Content` group with the permission `Addons:ContentReview`.
- Create a `Reviewers: Add-ons` group with the permission `Addons:PostReview`.
- Create a `Reviewers: Moderators` group with the permission `Ratings:Moderate`. That permission replaces the old rules for moderating user reviews.
- Remove the obsolete concept of `Limited Reviewers`.

Fix #6611

### CODE REVIEW

File: src/olympia/reviewers/models.py [hunk 1]
```diff
@@ -667,13 +669,14 @@ def get_leaderboards(cls, user, days=7, types=None, addon_type=None):
         leader_top = []
         leader_near = []
 
-        query = cls._leaderboard_query(since=week_ago, types=types,
-                                       addon_type=addon_type)
+        qs = cls._leaderboard_list(
```
Comment (eviljeff) [inline src/olympia/reviewers/models.py]: nit: `qs` is not a queryset - `_leaderboard_list` was changed to return a list.

File: src/olympia/reviewers/models.py [hunk 2]
```diff
@@ -1059,7 +1062,7 @@ def get_auto_approved_queue(cls):
 
     @classmethod
     def get_content_review_queue(cls):
-        """Return a queryset of Addon objects that have been auto-approved and
+        """Return a qsset of Addon objects that have been auto-approved and
```
Comment (eviljeff) [inline src/olympia/reviewers/models.py]: a `qsset`?

File: src/olympia/users/models.py [hunk 3]
```diff
@@ -180,6 +180,12 @@ def is_staff(self):
         from olympia.access import acl
         return acl.action_allowed_user(self, amo.permissions.ADMIN)
 
+    @property
+    def is_reviewer(self):
```
Comment (eviljeff) [inline src/olympia/users/models.py]: This function makes me uneasy - it's likely to be misused at some point, even though the comments (correctly) point out it shouldn't be used for ACL checks.  As far as I can see it's only used in one place (in live code) anyway - just move `self.groups.filter(name__startswith='Reviewers: ').exists()` there?

Comment (diox) [reply src/olympia/users/models.py]: Agreed

**Pull request merged**